src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -290,7 +290,7 @@ public static bool IsWindowsAzureManagedHosting(this IEnvironment environment)
         /// <returns><see cref="true"/> if running in Kubernetes environment; otherwise, false.</returns>
         public static bool IsAnyKubernetesEnvironment(this IEnvironment environment)
         {
-            return environment.IsKubernetesManagedHosting() || environment.IsManagedAppEnvironment();
+            return environment.IsKubernetesManagedHosting() || environment.IsManagedAppEnvironment() || environment.IsConnectedAppEnvironment();;
```

Comment:
The IsAnyKubernetesEnvironment is used in some other places including for example in [ScriptHostIdProvider](https://github.com/Azure/azure-functions-host/blob/c6365fdbba0f895b24da785a9b2f31ad0a759530/src/WebJobs.Script/Host/ScriptHostIdProvider.cs#L73), [setting hostname](https://github.com/Azure/azure-functions-host/blob/c6365fdbba0f895b24da785a9b2f31ad0a759530/src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs#L711), using [KubernetesEventGenerator](https://github.com/Azure/azure-functions-host/blob/c6365fdbba0f895b24da785a9b2f31ad0a759530/src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs#L119), adding [EnvironmentReadyCheckMiddleware](https://github.com/Azure/azure-functions-host/blob/c6365fdbba0f895b24da785a9b2f31ad0a759530/src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs#L41). Please ensure that these are paths that you would want for logic apps in ARC too.

Reply:
LogicApps specifically need it while setting hostname where FUNCTIONS_API_SERVER is set.
But looking it from infrastructure perpective, a connected environment in container apps is actually a Kubernetes Environment.

So its fine to consider that in IsAnyKubernetesEnvironment()